legateboost/models/tree.py

```diff
@@ -313,3 +315,124 @@ def __mul__(self, scalar: Any) -> "Tree":
+        # if the node is already leaf then its child can go off the end of the array
+        # use np.minimum to avoid this
+        nodes_trueleafs = self.is_leaf(
+            np.minimum(tree_max_nodes - 1, self.left_child(all_nodes_idx))
+        ).astype(int)
+        nodes_falsenodeids = self.right_child(all_nodes_idx)
+        nodes_falseleafs = self.is_leaf(
+            np.minimum(tree_max_nodes - 1, self.right_child(all_nodes_idx))
+        ).astype(int)
+
+        for output_idx in range(0, self.leaf_value.shape[1]):
+            leaf_targetids = np.full(self.feature.size, output_idx, dtype=np.int64)
+            leaf_weights = numpy_helper.from_array(
+                self.leaf_value[:, output_idx].__array__()
+            )
+
+            onnx_nodes.append(
+                onnx.helper.make_node(
+                    "TreeEnsemble",
+                    ["X"],
+                    ["pred" + str(output_idx)],
+                    domain="ai.onnx.ml",
+                    n_targets=self.leaf_value.shape[1],
+                    membership_values=None,
+                    nodes_missing_value_tracks_true=None,
+                    nodes_hitrates=None,
+                    aggregate_function=1,
+                    post_transform=0,
+                    tree_roots=[0],
+                    nodes_modes=make_tensor(
+                        "nodes_modes",
+                        onnx.TensorProto.UINT8,
+                        self.feature.shape,
+                        np.zeros_like(self.feature, dtype=np.uint8),
+                    ),
+                    nodes_featureids=nodes_featureids,
+                    nodes_splits=nodes_splits,
+                    nodes_truenodeids=nodes_truenodeids,
+                    nodes_trueleafs=nodes_trueleafs,
+                    nodes_falsenodeids=nodes_falsenodeids,
+                    nodes_falseleafs=nodes_falseleafs,
+                    leaf_targetids=leaf_targetids,
+                    leaf_weights=leaf_weights,
+                )
+            )
+
+            if output_idx == 0:
+                accumulated_pred = make_tensor_value_info(
+                    "accumulated_pred0", onnx.TensorProto.DOUBLE, [None, None]
+                )
+                onnx_nodes.append(
+                    onnx.helper.make_node(
+                        "Identity",
+                        ["pred" + str(output_idx)],
+                        ["accumulated_pred0"],
+                    )
+                )
+            else:
+                accumulated_pred = make_tensor_value_info(
+                    "accumulated_pred" + str(output_idx),
+                    onnx.TensorProto.DOUBLE,
+                    [None, None],
+                )
+                onnx_nodes.append(
+                    onnx.helper.make_node(
+                        "Add",
+                        [
+                            "accumulated_pred" + str(output_idx - 1),
+                            "pred" + str(output_idx),
```

Comment:
Could also use `f"pred{}" here as you use it in the other one, but doesn't matter.